src/esp/core/test/LoggingTest.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "esp/core/logging.h"
+
+#include <sstream>
+
+#include <Corrade/Containers/StaticArray.h>
+#include <Corrade/Containers/String.h>
+#include <Corrade/Containers/StringStl.h>
+#include <Corrade/Containers/StringView.h>
+#include <Corrade/TestSuite/Tester.h>
+
+namespace Cr = Corrade;
+
+namespace esp {
+
+namespace sim {
+namespace test {
+namespace {
+void debug(const Cr::Containers::StringView statement) {
+  ESP_DEBUG() << statement;
+}
+void warning(const Cr::Containers::StringView statement) {
+  ESP_WARNING() << statement;
+}
+}  // namespace
+}  // namespace test
+}  // namespace sim
+
+namespace gfx {
+namespace test {
+namespace {
+void debug(const Cr::Containers::StringView statement) {
+  ESP_DEBUG() << statement;
+}
+void warning(const Cr::Containers::StringView statement) {
+  ESP_WARNING() << statement;
+}
+}  // namespace
+}  // namespace test
+}  // namespace gfx
+
+namespace logging {
+namespace test {
+namespace {
+
+struct LoggingTest : Cr::TestSuite::Tester {
+  explicit LoggingTest();
+
+  void envVarTest();
+};
+
+constexpr const struct {
+  const char* envString;
+  const char* expected;
+} EnvVarTestData[]{
+    {nullptr,
+     "[Subsystem: Default] DebugDefault\n[Subsystem: Default] WarningDefault\n"
```

Comment:
The `Subsystem: ` prefix seems awfully verbose. I also wonder if `[Subsystem: Default]` is worth omitting altogether for brevity's sake.

Reply:
Yeah, I made it `[<SubsystemName>]` in my current working branch.